src/DynamicNLPModels.jl

```diff
@@ -603,24 +609,25 @@ function _build_dense_lq_dynamic_model(dnlp::LQDynamicData{T,V,M,MK}) where {T,
     num_real_bounds = sum(bool_vec)
 
     if num_real_bounds == length(sl)
-        J2         = block_B[(1 + ns):end,:]
+        J2         = (similar(Q, ns * N, nu * N) .= 0)
+        J2        .= block_B[(1 + ns):end,:]
         As0_bounds = As0[(1 + ns):end,1]
     else        
-        J2         = (similar(Q, (num_real_bounds * N, nu * N)) .= 0)
-        As0_bounds = (similar(Q, (num_real_bounds * N, 1)) .= 0)
+        J2         = (similar(Q, num_real_bounds * N, nu * N) .= 0)
+        As0_bounds = (similar(Q, num_real_bounds * N, 1) .= 0)
         for i in 1:N
             row_range = (1 + (i - 1) * num_real_bounds):(i * num_real_bounds)
-            J2[row_range, :] = block_B[(1 + ns * i): (ns * (i + 1)), :][bool_vec, :]
-            As0_bounds[row_range, :] = As0[(1 + ns * i):(ns * (i + 1)), :][bool_vec, :]
+            J2[row_range, :] .= block_B[(1 + ns * i): (ns * (i + 1)), :][bool_vec, :]
+            As0_bounds[row_range, :] .= As0[(1 + ns * i):(ns * (i + 1)), :][bool_vec, :]
         end
 
 
         sl = sl[bool_vec]
         su = su[bool_vec]
     end
 
-    lcon2 = (similar(Q, (size(J2, 1), 1)) .= 0)
-    ucon2 = (similar(Q, (size(J2, 1), 1)) .= 0)
+    lcon2 = (similar(Q, size(J2, 1), 1) .= 0)
```

Comment:
what happens if `num_real_bounds != length(sl)` and `J2` has not been instantiated on line 612?

Reply:
I think it would still be instantied in the `else` statement, but it is redundant. I agree that I should put it outside the `if` statements.